src/collaboration/Portal.ts

```diff
@@ -151,6 +152,14 @@ export class Portal {
 		case BroadcastType.MouseLocation:
 			this.collab.updateCursor(decoded.payload)
 			break
+		case BroadcastType.ViewportUpdate:
+			this.collab.updateCollaboratorViewport(
+				decoded.payload.userId,
```

Comment:
Just an idea, we could also pass the viewport size to the backend to record depending on the screens size of the user?